server/comms/comms_test.go

```diff
@@ -34,14 +37,51 @@ func newServer() *Server {
 	}
 }
 
+func giveItASecond(f func() bool) bool {
```

Comment:
Even though we are eliminating `time.Sleep` for synchronization, there are still some times where a channel isn't convenient or would add significant complexity. `giveItASecond` can be used to sync to some arbitrary condition, e.g. the client's presence in the server's client map, with a 1 ms delay between checks.